test/functional/wallet_signer.py

```diff
@@ -15,11 +15,16 @@
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.authproxy import JSONRPCException
 
 
 class WalletSignerTest(BitcoinTestFramework):
+    def add_options(self, parser):
+        parser.add_argument("--invalid", dest="invalid", default=False, action="store_true",
+                            help="Test invalid external signer")
+
     def mock_signer_path(self):
-        path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'mocks', 'signer.py')
+        path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'mocks', 'invalid_signer.py' if self.options.invalid else 'signer.py')
```

Comment:
Adding a new `invalid_signer.py` is fine with me, though an alternative approach would be to keep one mock, and then use the device fingerprint to decide the behavior.

In that case you would expand the `enumerate` mock to add a signer with e.g. `"fingerprint": "00000003"`, and then `getdescriptors` would return an invalid descriptor for `00000003`.

Reply:
I could get that to work, it seems that it is not supported yet ? (see https://github.com/bitcoin/bitcoin/blob/master/src/wallet/external_signer_scriptpubkeyman.cpp#L47) so instead I re-start the node and change signer path to point to the invalid signer mock.

Reply:
`createwallet` doesn't let you specify a fingerprint, so this workaround is probably fine for now.